corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -733,15 +734,20 @@ def __repr__(self):
 
     def render(self, context):
         if self.name not in context and self.value:
+            # Check that there isn't already an entry point from the other bundler tool
+            # If there is, don't add this one, because having both set will cause js errors
+            other_tag = "js_entry" if self.name == "requirejs_main" else "requirejs_main"
+            other_value = None
+            for context_dict in context.dicts:
+                if other_tag in context_dict:
+                    other_value = context_dict.get(other_tag)
+                    msg = f"Discarding {self.value} {self.name} value because {other_value} is using {other_tag}"
+                    soft_assert('jschweers@dimagi.com', notify_admins=False, send_to_ops=False)(False, msg)
             # set name in block parent context
-            other_name = "js_entry" if self.name == "requirejs_main" else "requirejs_main"
-            other_values = [d.get(other_name) for d in context.dicts if other_name in d]
-            if any(other_values):
-                raise TemplateSyntaxError(f"""
-                    Cannot use both {self.name} ({self.value}) and {other_name} ({other_values[0]})
-                """.strip())
-            context.dicts[-2]['use_js_bundler'] = True
-            context.dicts[-2][self.name] = self.value
+            if not other_value:
```

Comment:
Any chance that `other_value` could be false-ish here even if it came from `context_dict.get(other_tag)`?

Nitpicky, but it would be slightly more conservative to use a strong check for `None` here.
```suggestion
            if other_value is None:
```
Or alternatively, if it's possible that `context_dict[other_tag]` could return `None`, another approach would be to use a sentinel value that is not a well-known constant like `None`.
```py
other_value = NOT_SET = object()
for context_dict in context.dicts:
    ...
if other_value is NOT_SET:
```

Reply:
I don't think `other_value` can be falsy, because of the `if ... self.value` check that starts this function.